app_test.go

```diff
@@ -172,6 +172,44 @@ func TestOptions(t *testing.T) {
 	})
 }
 
+func TestTimeoutOptions(t *testing.T) {
+	const timeout = time.Minute
+	// Further assertions can't succeed unless the test timeout is greater than the default.
+	require.True(t, timeout > DefaultTimeout, "test assertions require timeout greater than default")
+
+	var started, stopped bool
+	assertCustomContext := func(ctx context.Context, phase string) {
+		deadline, ok := ctx.Deadline()
+		require.True(t, ok, "no %s deadline", phase)
```

Comment:
Using `require` here seems unsafe. It will call `t.Fatal` in case of failure
to kill the current goroutine, and with `app.Start`, that's not the test
goroutine.

Reply:
Good point, fixed.